requester.go

```diff
@@ -0,0 +1,352 @@
+	}
+
+	return h2
+}
+
+// Clone returns a deep copy of a Requester
+func (r *Requester) Clone() *Requester {
+	s2 := *r
+	s2.Header = cloneHeader(r.Header)
+	s2.Trailer = cloneHeader(r.Trailer)
+	s2.URL = cloneURL(r.URL)
+	s2.QueryParams = cloneValues(r.QueryParams)
+
+	return &s2
+}
+
+// Request returns a new http.Request
+func (r *Requester) Request(opts ...Option) (*http.Request, error) {
+	return r.RequestContext(context.Background(), opts...)
+}
+
+// RequestContext does the same as Request, but requires a context
+func (r *Requester) RequestContext(ctx context.Context, opts ...Option) (*http.Request, error) {
+	reqs, err := r.withOpts(opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	bodyData, ct, err := reqs.getRequestBody()
+	if err != nil {
+		return nil, err
+	}
+
+	urlS := ""
+	if reqs.URL != nil {
+		urlS = reqs.URL.String()
+	}
+
+	req, err := http.NewRequest(reqs.Method, urlS, bodyData)
+	if err != nil {
+		return nil, merry.Prepend(err, "creating request")
+	}
+
+	// if we marshaled the body, use our content type
+	if ct != "" {
+		req.Header.Set("Content-Type", ct)
+	}
+
+	if reqs.ContentLength != 0 {
+		req.ContentLength = reqs.ContentLength
+	}
+
+	if reqs.GetBody != nil {
+		req.GetBody = reqs.GetBody
+	}
+
+	if reqs.Host != "" {
+		req.Host = reqs.Host
+	}
+
+	req.TransferEncoding = reqs.TransferEncoding
+	req.Close = reqs.Close
+	req.Trailer = reqs.Trailer
+
+	for k, v := range reqs.Header {
+		req.Header[k] = v
+	}
+
+	if len(reqs.QueryParams) > 0 {
+		if req.URL.RawQuery != "" {
+			existingValues := req.URL.Query()
+
+			for key, value := range reqs.QueryParams {
+				for _, v := range value {
+					existingValues.Add(key, v)
+				}
+			}
+			req.URL.RawQuery = existingValues.Encode()
+		} else {
+			req.URL.RawQuery = reqs.QueryParams.Encode()
+		}
+	}
+
+	return req.WithContext(ctx), nil
+}
+
+// getRequestBody returns the io.Reader which should be used as the body of new Requester
+func (r *Requester) getRequestBody() (body io.Reader, contentType string, _ error) {
+	switch v := r.Body.(type) {
+	case nil:
+		return nil, "", nil
+	case io.Reader:
+		return v, "", nil
+	case string:
+		return strings.NewReader(v), "", nil
+	case []byte:
+		return bytes.NewReader(v), "", nil
+	default:
+		marshaler := r.Marshaler
+		if marshaler == nil {
+			marshaler = DefaultMarshaler
+		}
+
+		b, ct, err := marshaler.Marshal(r.Body)
+
+		if err != nil {
+			return nil, "", merry.Prepend(err, "marshaling body")
+		}
+
+		return bytes.NewReader(b), ct, nil
+	}
+}
+
+// Send executes a request with the Doer
+func (r *Requester) Send(opts ...Option) (*http.Response, error) {
+	return r.SendContext(context.Background(), opts...)
+}
+
+// withOpts is like With(), but skips the clone if there are no options to apply
+func (r *Requester) withOpts(opts ...Option) (*Requester, error) {
+	if len(opts) > 0 {
+		return r.With(opts...)
+	}
+
+	return r, nil
+}
+
+// SendContext does the same as Request, but requires a context
```

Comment:
I think it should be this? 
```suggestion
// SendContext does the same as Send, but requires a context
```